client/src/main/java/io/pravega/client/stream/impl/SegmentSelector.java

```diff
@@ -142,7 +151,7 @@ public Segment getSegmentForEvent(String routingKey) {
         currentSegments = currentSegments.withReplacementRange(sealedSegment, successors);
         createMissingWriters(segmentSealedCallback, currentSegments.getDelegationToken());
         log.debug("Fetch unacked events for segment: {}, and adding new segments {}", sealedSegment, currentSegments);
-        return writers.remove(sealedSegment).getUnackedEventsOnSeal();
+        return writers.get(sealedSegment).getUnackedEventsOnSeal();
```

Comment:
Please add a javadoc to this method explaining the impl and that remove needs to be called later.

Reply:
done.